sale_payment_mgmt/tests/test_access_rights.py

```diff
@@ -0,0 +1,103 @@
+                "inbound_payment_method_ids": [(6, 0, [cls.account_payment_method.id])],
+            }
+        )
+        # account_payment
+        cls.account_payment = cls.env["account.payment"].create(
+            {
+                "payment_type": "inbound",
+                "partner_type": "customer",
+                "partner_id": cls.res_partner.id,
+                "amount": 100,
+                "payment_date": fields.Date.from_string("2019-01-01"),
+                "communication": "payment_test",
+                "journal_id": cls.account_journal.id,
+                "payment_method_id": cls.account_payment_method.id,
+            }
+        )
+        # Users
+        Users = cls.env["res.users"].with_context(no_reset_password=True)
+        # sale_salesman
+        cls.user_sale_salesman = Users.create(
+            {
+                "name": "sale_salesman",
+                "login": "sale_salesman",
+                "email": "sale_salesman@example.com",
+                "groups_id": [
+                    (6, 0, [cls.env.ref("sales_team.group_sale_salesman").id])
+                ],
+            }
+        )
+        # sale_salesman_all_leads
+        cls.user_sale_salesman_all_leads = Users.create(
+            {
+                "name": "sale_salesman_all_leads",
+                "login": "sale_salesman_all_leads",
+                "email": "sale_salesman_all_leads@example.com",
+                "groups_id": [
+                    (6, 0, [cls.env.ref("sales_team.group_sale_salesman_all_leads").id])
+                ],
+            }
+        )
+        # sale_manager
+        cls.user_sale_manager = Users.create(
+            {
+                "name": "sale_manager",
+                "login": "sale_manager",
+                "email": "sale_manager@example.com",
+                "groups_id": [
+                    (6, 0, [cls.env.ref("sales_team.group_sale_manager").id])
+                ],
+            }
+        )
+
+    def test_access_sale_salesman(self):
+        self.assertIn(
+            self.account_payment.id,
+            self.env["account.payment"]
+            .with_user(self.user_sale_salesman)
+            .search([])
+            .ids,
+            "User with group sale_salesman should be able to read payments",
+        )
+
+    def test_access_sale_salesman_all_leads(self):
+        self.assertIn(
+            self.account_payment.id,
+            self.env["account.payment"]
+            .with_user(self.user_sale_salesman_all_leads)
+            .search([])
+            .ids,
+            "User with group sale_salesman_all_leads should be able to read payments",
+        )
+
+    def test_access_sale_manager(self):
+        self.assertIn(
+            self.account_payment.id,
+            self.env["account.payment"]
+            .with_user(self.user_sale_manager)
+            .search([])
+            .ids,
+            "User with group sale_manager should be able to read payments",
+        )
```

Comment:
Everything seems great! Just one thing, IMHO, you could add some more tests, for example, for when a User doesn't have proper rights, in which case he shouldn't be able to see the Payments. I don't know if it's that important though. What do you think @pedrobaeza ?

Reply:
Good idea, test improved now